src/Hl7.Fhir.Base/Model/Date.cs

```diff
@@ -38,41 +38,113 @@ namespace Hl7.Fhir.Model
     public partial class Date
     {
         public Date(int year, int month, int day)
-            : this(string.Format(FhirDateTime.FMT_YEARMONTHDAY, year, month, day))
+            : this(string.Format(System.Globalization.CultureInfo.InvariantCulture, FhirDateTime.FMT_YEARMONTHDAY, year, month, day))
         {
         }
 
         public Date(int year, int month)
-            : this(string.Format(FhirDateTime.FMT_YEARMONTH, year, month))
+            : this(string.Format(System.Globalization.CultureInfo.InvariantCulture, FhirDateTime.FMT_YEARMONTH, year, month))
         {
         }
 
-        public Date(int year) : this(string.Format(FhirDateTime.FMT_YEAR, year))
+        public Date(int year) : this(string.Format(System.Globalization.CultureInfo.InvariantCulture, FhirDateTime.FMT_YEAR, year))
         {
         }
 
+        public static Date FromDateTimeOffset(DateTimeOffset date) => new(date.Year, date.Month, date.Day);
+
         /// <summary>
-        /// Gets the current date in the local timezone
+        /// Gets the current date in the local timezone.
         /// </summary>
-        /// <returns>Gets the current date in the local timezone</returns>
-        public static Date Today() => new(DateTimeOffset.Now.ToString("yyyy-MM-dd"));
+        public static Date Today() => FromDateTimeOffset(DateTimeOffset.Now);
 
         /// <summary>
-        /// Gets the current date in the timezone UTC
+        /// Gets the current date in UTC.
         /// </summary>
-        /// <returns>Gets the current date in the timezone UTC</returns>
-        public static Date UtcToday() => new(DateTimeOffset.UtcNow.ToString("yyyy-MM-dd"));
+        public static Date UtcToday() => FromDateTimeOffset(DateTimeOffset.UtcNow);
+
+        [NonSerialized]  // To prevent binary serialization from serializing this field
+        private P.Date? _parsedValue = null;
+
+        private static readonly P.Date INVALID_VALUE = P.Date.Today();
```

Comment:
This would resolve to the current value on initialization of the app, hence long running services would have the boot time in this value, shouldn't it be a function so it is evaluated each time it is called?

Reply:
No. `hasInvalidParsedValue()` uses `ReferenceEquals()` to compare the cached time to INVALID_VALUE, so the actual value does not matter: it could have been mine or your birthdate as well. It's used here as a sentinel value to just indicate the value is invalid (as opposed to not cached (null)). It avoids keeping a separate bool `hasInvalidValue` that I need to keep synchronized.

Reply:
In case someone is looking at the data, would it be better to use something like DateTime.MinValue to make it obvious?